CHANGELOG.md

```diff
@@ -4,6 +4,38 @@
+* Fix leaking subscriptions in emitter builders and monix ops
+* Faster code with less memory allocations for constructing VNodeProxies, a lot of mutability but just internally :)
+* Remove implicit scheduler from builder dsl, we have a `SchedulerAction` modifier now, which gets the scheduler injected from the outer `Outwatch.render` call. We delay subscriptiuons in a `SchedulerAction` until a node is going to be inserted and use the same scheduler for the whole ui of the app.
+* Child commands to send an `Observable[ChildCommand]` that can incrementally update a list of nodes. Gives major improvments in speed as opposed to rerendering the whole list of nodes like a `Observable[Seq[VNode]]`.
+* Add thunk nodes. This is also useful for performance reasons, but you can keep writing your components as before with `Observable[Seq[VNode]]`. You can define a thunk node like this: `div.thunk("unique-component-name")(arg1, arg2)(VDomModifier(arg1, arg2, ...))`. Here `arg1` and `arg2` are the only dependencies of this div, so it only needs to be rerendered if one of these changes. So thunk delays the execution of the VDomModifier in the last parameter list (call-by-name) and only computes a new node if one of the arguments has changed. if the arguments stay the same, it can just reuse the node from the previous run (identified by the "unique-component-name"). For the perofmance this comes close to ChildCommands. It works similar to React components with a shouldRender function, which compares the previous and the next state.
+* Sync emitters so that stopPropagation/preventDefault actually work and are not behind an async boundary. Only async if needed.
+* Add `mapTo` to emitter builder to get the value to be mapped by-name.
+* Add `async` to enforce an async boundary in the emitter Observable.
+* Add `useLatest` to emitter builder for combining two emitter builders: e.g. `onClick.useLatest(onDomMount.asHtml).foreach { elem => elem.blur() }
+* Add `foreach` to emitter builder. foreach fits naturally with methods like `map`, `collect`, etc and gives a way to interact nicely with the dom: `onDomMount.asHtml.foreach { elem => elem.focus }`. The sideeffect factories are deprecated.
+* Userful emitter builder factories: `EmitterBuilder.fromObservable(Observable)` or `emitter(Observable)` and `EmitterBuilder.ofModifier`.
+* Less function allocations when handling hooks and emitters.
+* Do not return IO in Sink.create. The sink itself is not a sideeffect. If there is a sideeffect and therefore a reason for IO, then it lies in the function. Now then the function should be in IO beforehand. So, you will use it this way anyhow: `statefulAction.map(Sink.create _)`.
+* Add DomPreUpdate hook: `onDomPreUpdate`
+* Add Init hook, but it carries not element and is more of interest when working with the `VNodeProxy`, so you can only use it wirk InitHook(f: proxy => Unit).
```

Comment:
Oh yeah, that sentence makes no sense. I will correct it. What I meant is that you do not have init hooks in the dsl. There is no `onSnabbdomInit: EmitterBuilder[dom.Element, VDomModifier]` as you have with `onSnabbdomInsert` or `onSnabbdomDestroy`. This is because, when the `init` callback is called, the element of the snabbdom `VNodeProxy` is still null. So you only construct such a hook with `InitHook((proxy: VNodeProxy) => ???)`.

I will correct it.